### ISSUE TITLE
Using standard deviation or variance when sampling torch.distributions

### ISSUE THREAD

**Issue by thorinf** [MaximeRobeyns/bayesian_flow_networks#1]
Hi,

![image](https://github.com/MaximeRobeyns/bayesian_flow_networks/assets/91201457/04b5ff4d-556c-46b8-9947-c85ecb721f50)

I noticed in your implementation you are passing gamma * (1 - gamma) to torch.distributions https://github.com/MaximeRobeyns/bayesian_flow_networks/blob/9b20e89b301b55105393a444da658b0fc55c870d/torch_bfn/bfn.py#L126 and similarly in sampling
https://github.com/MaximeRobeyns/bayesian_flow_networks/blob/9b20e89b301b55105393a444da658b0fc55c870d/torch_bfn/bfn.py#L202

But when I looked at the docs for torch.distributions.Normal it seems to be that the standard deviation is the second arg, not the variance. 

I feel like this should be (gamma * (1 - gamma)).sqrt(). I've been struggling to get nice results with my continuous implementation, and yours are very good, so I definitely could be missing something

Thanks

**Comment by MaximeRobeyns** [comment]
Hi,

Thank you for the catch! Indeed, PyTorch expects a standard deviation, and passing this instead of the variance seems to lead to improved samples / training.

As a quick check, I re-ran the two moons example using the same hyperparameters, which yielded the following for the erroneous variance parametrisation:

![two_moons](https://github.com/MaximeRobeyns/bayesian_flow_networks/assets/17814194/33ce7b54-9884-47f3-b1d0-1ccf5bbddfc7)

And the following when using the standard deviation:

![two_moons_sqrt](https://github.com/MaximeRobeyns/bayesian_flow_networks/assets/17814194/41bebb7e-776e-4578-a2ff-fca394566510)

I feel like if your results look substantially less nice, then there could be something other than just this variance -> standard deviation difference...

Thanks!

**Comment by thorinf** [comment]
No problem.

Possibly sigma values. I've been working more on the discrete side for an LM task. I was actually very much hoping I was wrong about the sqrt, it would've impacted my discrete loss too.
